### PR TITLE
move so can send names to the action directly

### PR DESCRIPTION
**Pull request by hanbollar**
so we can add the names in our eng repo for easier addition by newer devs

----

lmk if there's anything else i need to change here // other updates - was doing this as a quick addition so mightve missed things

also if there's a better place to put the general maps or pass them through as parameters to the check function lmk ~

### CODE REVIEW

File: src/weekly-contributions.ts [hunk 1]
```diff
@@ -5,10 +5,13 @@ import {inspect} from 'util'
 type IssueStates = 'OPEN' | 'CLOSED'
 type PRStates = IssueStates | 'MERGED'
 
+const loginToNameMap: {[key: string]: string} = {}
```
Comment (Irev-Dev) [inline src/weekly-contributions.ts:8]:
```suggestion
let loginToNameMap: {[key: string]: string} = {}
```

File: src/weekly-contributions.ts [hunk 2]
```diff
@@ -5,10 +5,13 @@ import {inspect} from 'util'
...
+
 async function main() {
   const token = core.getInput('github-token')
   const dateStr = core.getInput('date')
   const markdownPrefix = core.getInput('markdown-prefix') || ''
+  const loginToNameMap = core.getInput('login-to-name-map') || ''
```
Comment (Irev-Dev) [inline src/weekly-contributions.ts:14]:
```suggestion
  loginToNameMap = JSON.parse(core.getInput('login-to-name-map')) || {}
```
remove const to reassign existing variable. Also github action inputs can only be strings, so the easiest way to get what we're after here is to use JSON and parse it on it's way it.
When adding the input in the github action, will have to be proper JSON with full `"` everywhere
```yml
login-to-name-map: |
  {
    "Irev-Dev": "Kurt",
    "hanbollar": "Hannah",
    "iterion": "Adam",
    "JBEmbedded": "JB",
    "jessfraz": "Jess",
    "jgomez720": "Josh",
    "JordanNoone": "Jordan",
    "mansoorsiddiqui": "Mansoor",
    "vonniwilliams": "Vonni"
  }
```

**Review by Irev-Dev — COMMENTED** [review]
Some things I think will likely be necessary for this to work.

**Comment by Irev-Dev** [comment]
Also you can try adding the new JSON variable here https://github.com/KittyCAD/ts-actions/blob/main/.github/workflows/weekly-contributions.yml#L20

File: src/weekly-contributions.ts (hunk 2 above)
Comment (hanbollar) [reply src/weekly-contributions.ts]: √

File: src/weekly-contributions.ts (hunk 1 above)
Comment (hanbollar) [reply src/weekly-contributions.ts]: √

**Comment by hanbollar** [comment]
@Irev-Dev is there anywhere else i'd have to add the `login-to-name-map` token for it to be recognized? getting https://github.com/KittyCAD/ts-actions/actions/runs/4358803930/jobs/7619877269#step:3:16 but matched the setup for items in weekly-contributions.ts

just checking if there's something off the top of your head, no worries if not, i can track down

**Comment by Irev-Dev** [comment]
@hanbollar, Hmm, I'm a bit confused myself, why this action didn't run
https://github.com/KittyCAD/ts-actions/blob/main/.github/workflows/weekly-contributions.yml#L5
The paths look correct to me 🤷  and you've made changes to those files 🤷 🤷

**Comment by hanbollar** [comment]
weird - it still shows same problem --> i'll followup with github support with the tagged ref number from https://github.com/KittyCAD/ts-actions/actions/runs/4359564604/jobs/7621488411#step:5:1189

**Comment by Irev-Dev** [comment]
Oh yeah checks are running, and they fail on [a143123](https://github.com/KittyCAD/ts-actions/pull/195/commits/a1431234087471d1eb05b2a60db65becfdd7a151), then CI made a commit to fix up the built js, and CI doesn't run after bot commit.

I made a dumby commit just to check.

Looks like the action does run but fails to make a comment on this PR, not sure exactly what the issue is.
https://github.com/KittyCAD/ts-actions/actions/runs/4359564604/jobs/7621488411

**Comment by hanbollar** [comment]
ah sounds good

spelunking into the support tickets i go ~

**Comment by hanbollar** [comment]
looks like it's all good now - 🤷

**Pull request merged**